File: backend/src/main/java/com/ams/entity/AssetModel.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("asset_model")
public class AssetModel implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;
    private String modelNo;
    private Long categoryId;
    private Long manufacturerId;
    private Long fieldsetId;
    private String specifications;
    private String description;

    private Integer status;
    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/assetModel.ts
import http from '@/utils/http';

export interface AssetModel {
  id?: number;
  name: string;
  modelNo?: string;
  categoryId?: number;
  manufacturerId?: number;
  fieldsetId?: number;
  specifications?: string;
  description?: string;
  status?: number;
  remark?: string;
}

export const getAssetModels = (params?: Record<string, any>) =>
  http.get<any>('/asset-models', { params });

export const getAssetModelOptions = () =>
  http.get<AssetModel[]>('/asset-models/options');

export const getAssetModelDetail = (id: number) =>
  http.get<AssetModel>(`/asset-models/${id}`);

export const createAssetModel = (data: AssetModel) =>
  http.post<AssetModel>('/asset-models', data);

export const updateAssetModel = (id: number, data: AssetModel) =>
  http.put<AssetModel>(`/asset-models/${id}`, data);

export const deleteAssetModel = (id: number) =>
  http.delete(`/asset-models/${id}`);
